src/baskerville/snps.py

```diff
@@ -128,54 +131,95 @@ def score_snps(params_file, model_file, vcf_file, worker_index, options):
 
     # setup output
     scores_out = initialize_output_h5(
-        options.out_dir, options.snp_stats, snps, targets_length, targets_strand_df
+        options.out_dir,
+        options.snp_stats,
+        snps,
+        targets_length,
+        targets_strand_df,
+        num_shifts,
     )
 
     # SNP index
     si = 0
 
     for sc in tqdm(snp_clusters):
         snp_1hot_list = sc.get_1hots(genome_open)
-
-        # predict reference
         ref_1hot = np.expand_dims(snp_1hot_list[0], axis=0)
-        ref_preds = seqnn_model(ref_1hot)[0]
-
-        # untransform predictions
-        if options.targets_file is not None:
-            if options.untransform_old:
-                ref_preds = dataset.untransform_preds1(ref_preds, targets_df)
-            else:
-                ref_preds = dataset.untransform_preds(ref_preds, targets_df)
-
-        # sum strand pairs
-        if strand_transform is not None:
-            ref_preds = ref_preds * strand_transform
 
-        for alt_1hot in snp_1hot_list[1:]:
-            alt_1hot = np.expand_dims(alt_1hot, axis=0)
-
-            # predict alternate
-            alt_preds = seqnn_model(alt_1hot)[0]
+        # predict reference
+        ref_preds = []
+        for shift in options.shifts:
+            ref_1hot_shift = dna.hot1_augment(ref_1hot, shift=shift)
+            ref_preds_shift = seqnn_model(ref_1hot_shift)[0]
 
             # untransform predictions
             if options.targets_file is not None:
                 if options.untransform_old:
-                    alt_preds = dataset.untransform_preds1(alt_preds, targets_df)
+                    ref_preds_shift = dataset.untransform_preds1(
+                        ref_preds_shift, targets_df
+                    )
                 else:
-                    alt_preds = dataset.untransform_preds(alt_preds, targets_df)
+                    ref_preds_shift = dataset.untransform_preds(
+                        ref_preds_shift, targets_df
+                    )
 
             # sum strand pairs
             if strand_transform is not None:
-                alt_preds = alt_preds * strand_transform
+                ref_preds_shift = ref_preds_shift * strand_transform
+
+            # save shift prediction
+            ref_preds.append(ref_preds_shift)
+        ref_preds = np.array(ref_preds)
+
+        ai = 0
+        for alt_1hot in snp_1hot_list[1:]:
+            alt_1hot = np.expand_dims(alt_1hot, axis=0)
+
+            # add compensation shifts for indels
+            indel_size = sc.snps[ai].indel_size()
+            if indel_size == 0:
+                alt_shifts = options.shifts
+            else:
+                # repeat reference predictions
+                ref_preds = np.repeat(ref_preds, 2, axis=0)
+
+                # add compensation shifts
+                alt_shifts = []
+                for shift in options.shifts:
+                    alt_shifts.append(shift)
+                    alt_shifts.append(shift - indel_size)
```

Comment:
This line adds shift in the direction opposite of the indel "direction", right? So if by default it is shifted to the right, this line will add a left shift

Reply:
Correct